### PR TITLE
QGrid component

### PR DESCRIPTION
**Pull request by ChxGuillaume**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+    <!-- TODO: 4 - Test your component here -->
+  </main>
   <q-hello-world-example />
+  <q-grid-example />
 </template>
 
 <script setup>
 import QHelloWorldExample from './components/QHelloWorld/Example.vue';
+import QGridExample from './components/QGrid/QGridExample.vue';
 </script>
```
Comment (MrZyr0) [inline src/App.vue]: On a dit qu'on ne mergais pas 🥹

File: src/components/QGrid/QCol.vue [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  if (props.sm) classes.push(parseInt(props.sm, 10) === 12 ? 'sm:w-full' : `sm:w-${props.sm}/12`);
+  if (props.md) classes.push(parseInt(props.md, 10) === 12 ? 'md:w-full' : `md:w-${props.md}/12`);
+  if (props.lg) classes.push(parseInt(props.lg, 10) === 12 ? 'lg:w-full' : `lg:w-${props.lg}/12`);
+  if (props.xl) classes.push(parseInt(props.xl, 10) === 12 ? 'xl:w-full' : `xl:w-${props.xl}/12`);
+
+  return classes.join(' ');
+});
+</script>
+
+<style lang="scss"></style>
```
Comment (MrZyr0) [inline src/components/QGrid/QCol.vue:46]: ça on peux le supprimer non ?

**Review by MrZyr0 — APPROVED** [review]

**Review by MrZyr0 — APPROVED** [review]

File: src/components/QGrid/QRow.vue [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+});
+
+const rowClasses = computed(() => {
+  const classes = [];
+
+  classes.push(`justify-${props.justify}`);
+  classes.push(`items-${props.align}`);
+
+  return classes.join(' ');
+});
```
Comment (MaxenceMottard) [inline src/components/QGrid/QRow.vue:28]:
```suggestion
const rowClasses = computed(() => [
  `justify-${props.justify}`,
  `items-${props.align}`
]);
```

File: src/components/QGrid/QRow.vue [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+</script>
+
+<style>
+.row {
+  width: 100%;
+}
```
Comment (MaxenceMottard) [inline src/components/QGrid/QRow.vue:34]: you can use `w-full` instead

File: src/components/QGrid/__tests__/QGrid.spec.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { describe, it, expect } from 'vitest'
+
+import { mount } from '@vue/test-utils'
+import HelloWorld from '../HelloWorld.vue'
+
+describe('HelloWorld', () => {
```
Comment (MaxenceMottard) [inline src/components/QGrid/__tests__/QGrid.spec.js:6]: You can delete this file if you didn't test your component 😅

File: src/components/index.js [hunk 6]
```diff
@@ -10,6 +12,8 @@ export default {
     app.component('QTabs', QTabs);
     app.component('QTab', QTab);
     app.component('QLoader', QLoader);
+    app.component('QRow', QRow);
+    app.component('QCol', QCol);
     // TODO: 3 - Declare your component here
   },
 };
```
Comment (MaxenceMottard) [inline src/components/index.js:19]: please rebase and add your components in exported object

File: src/components/QGrid/QGridExample.vue [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    <q-col class="bg-red-600" :cols="12" :sm="6" :md="4" :lg="3" :xl="2">test</q-col>
+  </q-row>
+</template>
+
+<script setup>
+import QRow from './QRow.vue';
+import QCol from './QCol.vue';
+</script>
+
+<style scoped></style>
```
Comment (MaxenceMottard) [inline src/components/QGrid/QGridExample.vue:21]:
```suggestion
```

File: src/components/QGrid/QCol.vue [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+const columnClasses = computed(() => {
+  const classes = [];
+
+  if (props.cols) classes.push(parseInt(props.cols, 10) === 12 ? 'w-full' : `w-${props.cols}/12`);
+  if (props.sm) classes.push(parseInt(props.sm, 10) === 12 ? 'sm:w-full' : `sm:w-${props.sm}/12`);
+  if (props.md) classes.push(parseInt(props.md, 10) === 12 ? 'md:w-full' : `md:w-${props.md}/12`);
+  if (props.lg) classes.push(parseInt(props.lg, 10) === 12 ? 'lg:w-full' : `lg:w-${props.lg}/12`);
+  if (props.xl) classes.push(parseInt(props.xl, 10) === 12 ? 'xl:w-full' : `xl:w-${props.xl}/12`);
+
+  return classes.join(' ');
```
Comment (MaxenceMottard) [inline src/components/QGrid/QCol.vue:42]:
```suggestion
  return classes;
```

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

File: src/components/QGrid/QGridExample.vue [hunk 9]
```diff
@@ -0,0 +1,21 @@
+<template>
```
Comment (MaxenceMottard) [inline src/components/QGrid/QGridExample.vue:1]: Could you rename this file as **Example.vue**

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

**Review by MaxenceMottard — APPROVED** [review]

**Review by MaxenceMottard — APPROVED** [review]

**Pull request merged**